grobid-core/src/main/java/org/grobid/core/jni/JEPThreadPool.java

```diff
@@ -55,7 +55,54 @@ private JEPThreadPool() {
         //executor = Executors.newFixedThreadPool(POOL_SIZE); 
         executor = Executors.newSingleThreadExecutor();
         // each of these threads is associated to a JEP instance
-        jepInstances = new HashMap<>();
+        jepInstances = new ConcurrentHashMap<>();
+    }
+
+    private Jep createJEPInstance() {
+        JepConfig config = new JepConfig();
+        Jep jep = null;
+        boolean success = false;
+        try {
+            File delftPath = new File(GrobidProperties.getInstance().getDeLFTFilePath());
+            if (!delftPath.exists()) {
+                throw new GrobidResourceException("DeLFT installation path does not exist");
+            }
+            if (!delftPath.isDirectory()) {
+                throw new GrobidResourceException("DeLFT installation path is not a directory");
+            }
+            config.addIncludePaths(delftPath.getAbsolutePath());
+            config.setClassLoader(Thread.currentThread().getContextClassLoader());
+            //System.out.println("jep instance thread: " + Thread.currentThread().getId());
+            jep = new Jep(config);
+            jepInstances.put(Thread.currentThread().getId(), jep);
+            // import packages
+            jep.eval("import os");
+            jep.eval("import numpy as np");
+            jep.eval("import keras.backend as K");
+            jep.eval("os.chdir('" + delftPath.getAbsolutePath() + "')");
+            jep.eval("from delft.utilities.Embeddings import Embeddings");
+            jep.eval("import delft.sequenceLabelling");
+            jep.eval("from delft.sequenceLabelling import Sequence");
+            jep.eval("from delft.sequenceLabelling.reader import load_data_and_labels_crf_file");
+            jep.eval("from delft.sequenceLabelling.reader import load_data_crf_string");
+            jep.eval("from sklearn.model_selection import train_test_split");
+            success = true;
+            return jep;
+        } catch(JepException e) {
+            LOGGER.error("JEP initialization failed", e);
+            throw new RuntimeException("JEP initialization failed", e);
```

Comment:
this was returning `null` before (after logging the error). Probably best if it throws an exception.